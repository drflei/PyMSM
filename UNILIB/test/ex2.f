      PROGRAM example2
C
      INCLUDE 'structure.h'
C
      INTEGER*4    kunit, kinit, ifail, kint, kext, nfbm, i, noprint
      CHARACTER*32 lbint, lbext
      REAL*8       year, param(10), alpha
      REAL*8       fbm, flm, fkm, fsm, fbeq, fs
      TYPE(ZDAT) mdate
      TYPE(ZGEO) mgeod, mpos
C
C     initialisation
C
      DATA kunit, kinit, kint, kext, nfbm, noprint/ 6, 1, 0, 5, 1, -1/
      DATA year, param, alpha/ 1995.0d0, 10*0.0d0, 90.0d0/
C     Note: param is not used when kext=5
C
C     date and time
      mdate%iyear   = 1995
      mdate%imonth  =    1
      mdate%iday    =    1
      mdate%ihour   =    0
      mdate%imin    =    0
      mdate%secs    =    0.0d0
C
C     geodetic location
      mgeod%radius  =20000.0d0 + 6371.2d0
      mgeod%colat   =  120.0d0
      mgeod%elong   =   30.0d0        
C
      CALL UT990 (kunit, kinit, ifail)
      IF( ifail .LT. 0 )STOP
C
      CALL UM510 (kint, year, lbint, kunit, ifail)
      IF( ifail .LT. 0 )STOP
C
      CALL UT540 (mdate)          
C
      CALL UM520 (kext, mdate%amjd, param,
     :           lbext, kunit, ifail)
      IF( ifail .LT. 0 )STOP
C
C     convert the location from geodetic to geocentric coordinates
C
      CALL UM536 (mgeod, mpos)
C
C     loop on the hours
C
      WRITE(kunit,1000)
C
      DO i=0, 23
C
C       update the Sun position and the SM coordinates
C
        mdate%ihour =    i
        CALL UT540 (mdate)          
C
        CALL UM522 (mdate%amjd, noprint)
        CALL UM524 (noprint)
C
C       evaluate the (B, L) coordinates
C
        CALL UL220 (mpos, alpha, nfbm, fbm, flm, fkm, fsm, fbeq, 
     :              fs, ifail)
        IF( ifail .LT. 0 )STOP
C
C       write the result
C
        WRITE(kunit,1001) i, fbm, flm
C
      ENDDO
C
      WRITE(kunit,1002)
C
 1000 FORMAT(
     :  /25('-'),/'RESULT: Magnetic field intensity at mirror points ',
     :   ' and McIlwain shell parameter',
     :  /4x,'hour',15x,'intensity',7x,'shell parameter')
 1001 FORMAT( 4x,i4,2es24.14 )
 1002 FORMAT( /25('-') )
C
      END

