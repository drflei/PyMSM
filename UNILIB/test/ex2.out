
                                    SPENVIS
                                 UNILIB Library
                                 Version 2.22_YCH_initPrep
                           Generated by SPENVIS team


 --- Main control parameters ---

      UC160 (general constants):
                      Geoid major axis = 6378.16     km
                      Geoid minor axis = 6356.77     km
                Mean Earth radius (Re) = 6371.20     km
                  Ecliptic inclination =   23.441512 deg
                 McIlwain Earth dipole =    0.311653 G/Re^3

      UC190 (field line tracing):
                     Maximum step size =  477.8      km at one Re
              Maximum "last" step size =   15.0      km
                     Minimum step size =    2.0      km
                    Altitude precision =  200.       m
              Magnetic field precision =    0.0006   %

      UC192 (magnetic field evaluation):
             Max. geomagnetic latitude =   85.0      deg
                        Minimum radius =    0.157    Re
            (drift shell tracing):
                        Precision on L =    0.10000  %
            Precision on the longitude =    0.05     deg

 --- Geomagnetic field model ---

      Model ( 0): DGRF 1995                          Epoch 1995.
                             Order + 1 =   11
                     Calculation epoch = 1995.0      year
         Colatitude of the dipole pole =   10.68     deg
          Longitude of the dipole pole =  -71.42     deg
                   Earth dipole moment =    0.302151 G/Re^3
          Correction for the SAA drift =    0.00     deg

 --- External magnetic field model ---

      Model ( 5): Olson-Pfitzer 1977              
           Planetary activity index Kp =    0.0      (magnetopause check)

                                  Date =  01-Jan-1995
                        Universal Time =    0.00     hrs
            Greenwich angle of equinox =  100.18     deg
           GEO direction of the Sun, X =   -0.9200
                                     Y =   -0.0129
                                     Z =   -0.3917
                     Dipole tilt angle =  -25.91     deg

-------------------------
RESULT: Magnetic field intensity at mirror points  and McIlwain shell parameter
    hour               intensity       shell parameter
       0    5.69774556025675E-03    6.44007693668516E+00
       1    5.68399820899698E-03    6.41389680900312E+00
       2    5.66292214425666E-03    6.32303625248843E+00
       3    5.63894247811410E-03    6.21309917466108E+00
       4    5.61459559100788E-03    6.13508047733296E+00
       5    5.59192802270609E-03    6.09113862835756E+00
       6    5.57266307758409E-03    6.04693473516862E+00
       7    5.55658684988412E-03    5.99285569182935E+00
       8    5.54278038765815E-03    5.95386402228482E+00
       9    5.53152545671420E-03    5.94370911313331E+00
      10    5.52378025336341E-03    5.94768432021102E+00
      11    5.52002599772832E-03    5.95121889821842E+00
      12    5.52030423453167E-03    5.95882446333830E+00
      13    5.52481162472806E-03    5.98434505719205E+00
      14    5.53391261730843E-03    6.02690555798865E+00
      15    5.54743998878423E-03    6.07065413771089E+00
      16    5.56429681764972E-03    6.10393010634963E+00
      17    5.58312079429520E-03    6.13860055439532E+00
      18    5.60330044695083E-03    6.19887049004449E+00
      19    5.62504697350314E-03    6.28573555366209E+00
      20    5.64822141533409E-03    6.37033797233338E+00
      21    5.67104712345804E-03    6.41505631921816E+00
      22    5.68979460495423E-03    6.42108808963482E+00
      23    5.69975149975544E-03    6.42729159167912E+00

-------------------------
